### PR TITLE
sscs-9874-common-version-update

### PR DESCRIPTION
**Pull request by bensonlievoco**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-9874

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by ikirsanov — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java [hunk 1]
```diff
@@ -449,28 +450,38 @@ public void appealWithAppellantAndMultipleOtherPartiesAndMultipleFurtherEvidence
 
         assertEquals("Test Rep", pdfDocumentRequest.getAllValues().get(0).getData().get("name"));
         assertEquals("609-97-template (original sender)", documentCaptor.getAllValues().get(0).get(0).getName());
-        assertEquals("evidence-document2", documentCaptor.getAllValues().get(0).get(1).getName());
-        assertEquals("evidence-document", documentCaptor.getAllValues().get(0).get(2).getName());
+        Assertions.assertThat(documentCaptor.getAllValues().get(0))
```
Comment (dattipoeHMCTS) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java:453]: You're still hardcoding indexes, you could move this up to find "Test Rep" in all the values and so on

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java [hunk 2]
```diff
@@ -447,30 +448,40 @@ public void appealWithAppellantAndMultipleOtherPartiesAndMultipleFurtherEvidence
         assertEquals(3, documentCaptor.getAllValues().get(3).size());
         assertEquals(3, documentCaptor.getAllValues().get(4).size());
```
Comment (TeaWithLucas) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java:449]: These and some of the changes you made below can be covered by:
```
        Assertions.assertThat(documentCaptor.getAllValues())
            .hasSize(5)
            .extracting(ArrayList::new)
            .allSatisfy(pdfList -> Assertions.assertThat(pdfList)
                .hasSize(3)
                .extracting(Pdf::getName)
                .contains("evidence-document2", "evidence-document")
            )
```

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java [hunk 3]
```diff
@@ -29,9 +29,9 @@ private void verifyEvidenceIssued() {
...
         log.info("verifyEvidenceIssued ccdCaseId " + ccdCaseId);
 
-        assertNull(docs.get(0).getValue().getEvidenceIssued());
-        assertEquals("Yes",docs.get(1).getValue().getEvidenceIssued());
-        assertEquals("Yes",docs.get(2).getValue().getEvidenceIssued());
-        assertEquals("Yes",docs.get(3).getValue().getEvidenceIssued());
+        Assertions.assertThat(docs)
+            .extracting(SscsDocument::getValue)
+            .extracting(SscsDocumentDetails::getEvidenceIssued)
+            .contains("Yes", "Yes", "Yes");
```
Comment (TeaWithLucas) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java:35]: .contains("Yes", "Yes", "Yes")
can be simplied/made more elegant by doing:
.hasSize(3)
.containsOnly("Yes)

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java [hunk 4]
```diff
@@ -48,32 +48,33 @@ private void verifyEvidenceIsNotIssued() {
...
         List<SscsDocument> docs = caseData.getSscsDocument();
 
-        assertNull(docs.get(0).getValue().getEvidenceIssued());
-        assertEquals("Yes", docs.get(1).getValue().getEvidenceIssued());
-        assertEquals("Yes", docs.get(2).getValue().getEvidenceIssued());
-        assertEquals("Yes", docs.get(3).getValue().getEvidenceIssued());
+        Assertions.assertThat(docs)
+            .extracting(SscsDocument::getValue)
+            .extracting(SscsDocumentDetails::getEvidenceIssued)
+            .contains("Yes", "Yes", "Yes");
```
Comment (TeaWithLucas) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:65]: .contains("Yes", "Yes", "Yes")
can be simplied/made more elegant by doing:
.hasSize(3)
.containsOnly("Yes")

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java [hunk 5]
```diff
@@ -48,32 +48,33 @@ private void verifyEvidenceIsNotIssued() {
...
```
Comment (TeaWithLucas) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:76]: .contains("Yes", "Yes", "Yes")
can be simplied/made more elegant by doing:
.hasSize(3)
.containsOnly("Yes")

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java [hunk 6]
```diff
@@ -83,10 +84,10 @@ private void verifyEvidenceIssuedAndExistingReasonableAdjustmentRaised() {
...
```
Comment (TeaWithLucas) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:90]: .contains("Yes", "Yes", "Yes")
can be simplied/made more elegant by doing:
.hasSize(3)
.containsOnly("Yes")

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java [hunk 7]
```diff
@@ -48,32 +48,33 @@ private void verifyEvidenceIsNotIssued() {
...
         assertEquals("failedSendingFurtherEvidence", caseData.getHmctsDwpState());
         List<SscsDocument> docs = caseData.getSscsDocument();
-        assertNull(docs.get(0).getValue().getEvidenceIssued());
-        assertEquals("No", docs.get(1).getValue().getEvidenceIssued());
-        assertEquals("No", docs.get(2).getValue().getEvidenceIssued());
-        assertEquals("No", docs.get(3).getValue().getEvidenceIssued());
+        Assertions.assertThat(docs)
+            .extracting(SscsDocument::getValue)
+            .extracting(SscsDocumentDetails::getEvidenceIssued)
+            .contains("No", "No", "No");
```
Comment (TeaWithLucas) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:54]: .contains("Yes", "Yes", "Yes")
can be simplied/made more elegant by doing:
.hasSize(3)
.containsOnly("No")

**Review by TeaWithLucas — DISMISSED** [review]

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java (hunk 3 above)
Comment (bensonlievoco) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java:35]: your suggestion will limit the list of size is 3, it slightly change the original logic. The change is aim to check without the ordering but it is not about the size of the list. 

The code just have 3 Yes value. I think it is acceptable.

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java (hunk 2 above)
Comment (bensonlievoco) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java:449]: we still need to check the list contain  "evidence-document" and "evidence-document2". Use my coding can make sure the pair of the pdfDocumentRequest and documentCaptor consist and no change the original logic

Comment (bensonlievoco) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java:449]: I know this is good suggest, but we are aiming to dismiss the blocking of the ordering check, not restructuring the checking

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java (hunk 3 above)
Comment (filippo-possenti-version1) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ReissueAppellantAppointeeFurtherEvidenceHandlerFunctionalTest.java:35]: I'll make a call here and say that while Lucas' suggestion at first glance appears to be marginally better in this particular context, it's still subjective and I'm happy to keep the current implementation.

I'll mark this conversation as resolved.

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java (hunk 4 above)
Comment (filippo-possenti-version1) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:65]: I'll make a call here and say that while Lucas' suggestion at first glance appears to be marginally better in this particular context, it's still subjective and I'm happy to keep the current implementation.

I'll mark this conversation as resolved.

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java (hunk 5 above)
Comment (filippo-possenti-version1) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:76]: I'll make a call here and say that while Lucas' suggestion at first glance appears to be marginally better in this particular context, it's still subjective and I'm happy to keep the current implementation.

I'll mark this conversation as resolved.

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java (hunk 6 above)
Comment (filippo-possenti-version1) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:90]: I'll make a call here and say that while Lucas' suggestion at first glance appears to be marginally better in this particular context, it's still subjective and I'm happy to keep the current implementation.

I'll mark this conversation as resolved.

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java (hunk 7 above)
Comment (filippo-possenti-version1) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/IssueFurtherEvidenceHandlerFunctionalTest.java:54]: I'll make a call here and say that while Lucas' suggestion at first glance appears to be marginally better in this particular context, it's still subjective and I'm happy to keep the current implementation.

I'll mark this conversation as resolved.

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java (hunk 2 above)
Comment (filippo-possenti-version1) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/service/IssueFurtherEvidenceServiceIt.java:449]: I actually prefer Benson's approach in this particular case.

Surely the test code will be shorter with Lucas' approach but I learned over time that (counter-intuitively) there's value  on occasion in the tests' code being somewhat repetitive. In this particular case I had to think a lot less to understand Benson's code.

I'll make a judgment call here and say we'll keep this test as written by Benson.

I'll mark this conversation as resolved.

**Review by filippo-possenti-version1 — APPROVED** [review]

**Pull request merged**